pydis_site/apps/content/resources/guides/python-guides/subclassing_context.md

```diff
@@ -0,0 +1,137 @@
+
+class PromptView(discord.ui.View):
+    def __init__(self, *, timeout: float, author_id: int, ctx: commands.Context, delete_after: bool) -> None:
+        super().__init__(timeout=timeout)
+        self.value: typing.Optional[bool] = None
+        self.delete_after: bool = delete_after
+        self.author_id: int = author_id
+        self.ctx: Context = ctx
+        self.message: typing.Optional[discord.Message] = None
+
+    async def interaction_check(self, interaction: discord.Interaction) -> bool:
+        if interaction.user and interaction.user.id == self.author_id:
+            return True
+        else:
+            await interaction.response.send_message('This confirmation dialog is not for you.', ephemeral=True)
+            return False
+    # ensure that other members do not confirm or deny
+
+    async def on_timeout(self) -> None:
+        if self.delete_after and self.message:
+            await self.message.delete()
+        #delete the message if there is no response
+
+    @discord.ui.button(label='Confirm', style=discord.ButtonStyle.green)
+    async def confirm(self, button: discord.ui.Button, interaction: discord.Interaction):
+        self.value = True #returns True if the confirm button is pressed
+        await interaction.response.defer()
+        if self.delete_after:
+            await interaction.delete_original_message()
+        self.stop() #stops the view and deletes the confirmation message
+
+    @discord.ui.button(label='Cancel', style=discord.ButtonStyle.red)
+    async def cancel(self, button: discord.ui.Button, interaction: discord.Interaction):
+        self.value = False #returns False if the cancel button is pressed
+        await interaction.response.defer()
+        if self.delete_after:
+            await interaction.delete_original_message()
+        self.stop() #stops the view and deletes the confirmation
+
+# We have made two buttons, one for Confirm and one for Cancel.
+
+class CustomContext(commands.Context):
+    # create new Context methods here. For example:
+
+    async def send_embed(self, message, **kwargs):
+        return await self.send(embed=discord.Embed(description=message, colour=discord.Color.blurple(), **kwargs))
+        # This is a shortcut method to sending a custom embed. Pre-set values can be put here, for example the color kwarg in the example above means it has a pre-set color of blurple
+
+    async def send_error_embed(self, message, **kwargs):
+        return await self.send(embed=discord.Embed(description=message, colour=discord.Color.red(), **kwargs))
+        # embed with pre-set color of red
+
+
+    async def send_sucess_embed(self, message, **kwargs):
+        return await self.send(embed=discord.Embed(description=message, colour=discord.Color.green(), **kwargs))
+        # embed with pre-set color of green
+
+    # our custom prompt function using buttons  
```

Comment:
The comments here are probably better suited to be docstrings, since it's useful for the call site to know about.